sound/hda/intel-sdw-acpi.c

```diff
@@ -166,7 +166,7 @@ int sdw_intel_acpi_scan(acpi_handle *parent_handle,
 
 	info->handle = NULL;
 	status = acpi_walk_namespace(ACPI_TYPE_DEVICE,
-				     parent_handle, 1,
+				     parent_handle, 2,
```

Comment:
why two and why not three then?

we've got to provide more information here and explain what the controller might depend on.

I also wonder if this actually works, because in a number of cases we assume that the parent of the SoundWire controller is the PCI device.

Reply:
@plbossart This patch works. The only block without this patch is that it can't find the correct SoundWIre controller in ACPI table, which leads to fail to find the soundwire properties setting in the ACPI. The HW (PCI hierarchical structure) should not changed. The old BIOS works fine. We updated the BIOS and found this issue on the same board.

Reply:
@libinyang The commit should explain why 2 is the correct one. Ideally we could refer to a spec, but if this is based on a practical test, we should state that. What is between the PCI parent and the controller on this device? That might help to write a more understandable explanation.